ui/src/App.js

```diff
@@ -53,56 +90,257 @@ function App() {
   };
 
+  const sendMnemonic = async (mnemonic) => {
+    console.log("[INFO]: Send mnemonic to signer server.");
+    return await invoke('send_mnemonic', { mnemonic: mnemonic })
+  }
+
   const stopPasswordPrompt = async () => {
     console.log("[INFO]: Stop password prompt.");
     await invoke('stop_password_prompt');
   };
 
+  const resetAccount = async () => {
+    console.log("[INFO]: Resetting Account.");
+
+    await invoke('ui_disconnected');
+    await invoke('reset_account', { delete: true });
+
+  }
+
+  const restartServer = async (loginPage = false) => {
+    console.log("[INFO]: Restarting Server.");
+    await invoke('ui_disconnected');
+    await invoke('reset_account', { delete: false });
+
+    if (loginPage) {
+      setCurrentPage(LOGIN_PAGE);
+    } else {
+      setCurrentPage(CREATE_OR_RECOVER_PAGE);
+    }
+  }
+
   const endInitialConnectionPhase = async () => {
     console.log("[INFO]: End Initial Connection Phase");
     setIsConnected(true);
     hideWindow();
+    if (activeListeners.tx || activeListeners.reset_tray) return;
     listenForTxAuthorizationRequests();
+    listenForResetTrayRequests();
+    listenForShowSecretPhraseRequests();
+    setActiveListeners({
+      ...activeListeners,
+      tx: true,
+      reset_tray: true,
+      show_secret_phrase: true
+    })
   };
 
+  const endConnection = async () => {
+    console.log("[INFO]: Ending connection.");
+    setIsConnected(false);
+  }
+
+  const startCreate = async () => {
+    console.log("[INFO]: Start wallet creation process.")
+    setCurrentPage(CREATE_ACCOUNT_PAGE);
+  }
+
+  const startRecover = async () => {
+    console.log("[INFO]: Start recovery process.")
+    setCurrentPage(RECOVER_PAGE);
+  }
+
+  const cancelReset = async () => {
+    console.log("[INFO]: Cancel reset process.")
+    hideWindow();
+  }
+
+  const cancelSign = async () => {
+    console.log("[INFO]: Cancelling signing current transaction.");
+    await invoke('cancel_sign');
+  }
+
+  const endExportPhrase = async () => {
+    console.log("[INFO]: Ending export recovery phrase process.")
+    setExportingPhrase(false);
+    setExportedSecretPhrase(null);
+    hideWindow();
+  }
+
+  const getReceivingKeys = async () => {
+    const newReceivingKeys = await invoke('receiving_keys');
+    const newReceivingKey = newReceivingKeys[0];
+    setReceivingKey(newReceivingKey);
+    const newReceivingKeyDisplay = newReceivingKey ?
+      `${newReceivingKey.slice(0, 10)}...${newReceivingKey.slice(-10)}` :
+      '';
+    setReceivingKeyDisplay(newReceivingKeyDisplay);
+  }
+
+  // keeps the connect listener in sync with currentPage state
+  useEffect(() => {
+    currentPageRef.current = currentPage;
+  }, [currentPage])
+
+  // keeps show secret phrae listener in sync with exportingPhrase state
+  // whether or not we are currently exporting the phrase.
+  useEffect(() => {
+    exportingPhraseRef.current = exportingPhrase;
+  },[exportingPhrase])
+
```

Comment:
I would move these to the top of the component, right below all of our state, since it's also basic state-management logic.